src/PowerShellEditorServices/Services/DebugAdapter/Handlers/ConfigurationDoneHandler.cs

```diff
@@ -102,48 +102,36 @@ public Task<ConfigurationDoneResponse> Handle(ConfigurationDoneArguments request
 
         private async Task LaunchScriptAsync(string scriptToLaunch)
         {
-            // Is this an untitled script?
+            PSCommand cmd;
             if (ScriptFile.IsUntitledPath(scriptToLaunch))
             {
                 ScriptFile untitledScript = _workspaceService.GetFile(scriptToLaunch);
-
                 if (BreakpointApiUtils.SupportsBreakpointApis(_runspaceContext.CurrentRunspace))
                 {
-                    // Parse untitled files with their `Untitled:` URI as the file name which will cache the URI & contents within the PowerShell parser.
-                    // By doing this, we light up the ability to debug Untitled files with breakpoints.
-                    // This is only possible via the direct usage of the breakpoint APIs in PowerShell because
+                    // Parse untitled files with their `Untitled:` URI as the file name which will
+                    // cache the URI and contents within the PowerShell parser. By doing this, we
+                    // light up the ability to debug Untitled files with breakpoints. This is only
+                    // possible via the direct usage of the breakpoint APIs in PowerShell because
                     // Set-PSBreakpoint validates that paths are actually on the filesystem.
-                    ScriptBlockAst ast = Parser.ParseInput(untitledScript.Contents, untitledScript.DocumentUri.ToString(), out Token[] tokens, out ParseError[] errors);
-
-                    // This seems to be the simplest way to invoke a script block (which contains breakpoint information) via the PowerShell API.
-                    //
-                    // TODO: Fix this so the added script doesn't show up.
-                    var cmd = new PSCommand().AddScript(". $args[0]").AddArgument(ast.GetScriptBlock());
-                    await _executionService
-                        .ExecutePSCommandAsync<object>(cmd, CancellationToken.None, s_debuggerExecutionOptions)
-                        .ConfigureAwait(false);
+                    ScriptBlockAst ast = Parser.ParseInput(
```

Comment:
This whole thing is pointless if we decided to break line breakpoints for untitled files.

Reply:
@SeeminglyScience Do you think we could send a `ScriptBlock` type down the pipeline executor (much like `PSCommand`) to eventually `.Invoke()` it?